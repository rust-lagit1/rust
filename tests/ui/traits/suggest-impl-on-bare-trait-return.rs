//@ edition:2021
trait Trait {}

async fn fun() -> Trait {  //~ ERROR expected a type, found a trait
    todo!()
}

fn main() {}
