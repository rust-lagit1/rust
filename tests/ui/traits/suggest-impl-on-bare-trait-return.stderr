error[E0782]: expected a type, found a trait
  --> $DIR/suggest-impl-on-bare-trait-return.rs:4:13
   |
LL | fn fun() -> Trait {
   |             ^^^^^
   |
help: use `impl Trait` to return an opaque type, as long as you return a single underlying type
   |
LL | fn fun() -> impl Trait {
   |             ++++
help: alternatively, you can return an owned trait object
   |
LL | fn fun() -> Box<dyn Trait> {
   |             +++++++      +

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0782`.
